### PR TITLE
Nova chance: Remove jadelet, remove a ton of model code

### PR DESCRIPTION
**Pull request by GregWeil**
- react error page
- react layout
- remove the reactlet shim
- removed all model editing funcs since they were unused

exciting changes
- it doesn't flicker the login button on page load any more
- the 'for over 18 years' bit on the index page will update within an hour of [the day fog creek was announced](https://www.joelonsoftware.com/2000/08/14/20000814/). but it now says 'for over 17 years' because it went from 'since jan 1 2000' to 'since aug 14 2000'
- project models have like 6 lines of code outside of asprops

til
- `new Date(y,m,d)` has 0 indexed months but 1 indexed days? why

### CODE REVIEW

**Comment by GregWeil** [comment]
did some cleanup on the 'for x years' component, now updates every five minutes, and now uses purecomponent so it shouldn't try to re-render except when things actually change

File: shrinkwrap.yaml [hunk 1]
```diff
@@ -1579,16 +1544,6 @@ packages:
       node: '>=0.10.0'
     resolution:
       integrity: sha1-DQcLTQQ6W+ozovGkDi7bPZpMz3c=
-  /coffee-script/1.7.1:
```
Comment (judeallred) [inline shrinkwrap.yaml]: 🎉

File: src/presenters/pages/error.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const logo = "https://cdn.glitch.com/2bdfb3f8-05ef-4035-a06e-2043962a3a13%2Fcarp.svg";
+
+const drawStar = (canvas, context, color) => {
+  const x = Math.random() * canvas.width;
+  const y = Math.random() * canvas.height;
+  context.fillStyle = color;
+  context.fillRect(x, y, 2, 2);
+};
+
+class Stars extends React.Component {
```
Comment (judeallred) [inline src/presenters/pages/error.jsx]: neat, i learned some [react stuff](https://reactjs.org/docs/refs-and-the-dom.html) by reading this

File: src/presenters/pages/index.jsx [hunk 3]
```diff
@@ -60,22 +61,42 @@ class WhatIsGlitch extends React.Component {
...
+  }
+  
+  componentDidMount() {
+    this.timer = window.setInterval(() => {
+      this.setState({age: this.age()});
+    }, moment.duration(5, 'minutes').asMilliseconds());
+  }
+  
+  componentWillUnmount() {
+    window.clearTimeout(this.timer);
```
Comment (judeallred) [inline src/presenters/pages/index.jsx]: should be clearInterval, no?

File: src/presenters/pages/project.jsx [hunk 4]
```diff
@@ -133,52 +129,48 @@ const ProjectPage = ({
...
+        )}
+      </ProjectEditor>
+    ) : <NotFound name={name}/>}
+  </DataLoader>
 );
 ProjectPageLoader.propTypes = {
   name: PropTypes.string.isRequired,
 };
 
-// Let's keep layout in jade until all pages are react
```
Comment (judeallred) [inline src/presenters/pages/project.jsx]: this deleted line is my favorite deleted line

**Pull request merged**